### ISSUE TITLE
Site 63- Billerica Outbound - Account Ledger for Microsol and Mindray

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#881]
In GitLab by @kdjstudios on May 7, 2018, 08:27

**Submitted by:** "Cindy Pesa" <cindy.pesa@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2018-05-03-33174/conversation

**Server:** Internal

**Client/Site:** 63

**Account:** Multiple

**Issue:** 

The account ledger balance at the bottom of the ledger is not matching the number at the top of the ledger.

 

Here are the two accounts

![image](/uploads/7a6ce9e9d9a2fa9f4125e2ac984921a8/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 9, 2018, 08:38

@smasih @pchaudhary @tim.traylor @thomas.gentry 

Cindy has updated this ticket that it is priority.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 10, 2018, 01:18

Hi [Kyle] ( @kdjstudios ),  [Tim] ( @tim.traylor )
> The account ledger balance at the bottom of the ledger is not matching the number at the top of the ledger.

* We have a little confusion regarding the term "*bottom & top*". However, we guess that Cindy is talking about the ledger balance miscalculation. If that is the case then here are our findings from the investigation - 

* The very first invoice for both of the accounts gets wrongly calculated due to which the carry total of future invoices is got wrong.

* We also did check with the old ledger calculation & the ledger balance is same for the old one too.


![Screenshot](/uploads/09c434275c9fb482144bc30f2ff2d013/Screenshot.png)
----  
![Screenshot-2](/uploads/fc5d067ab0d23ca1a575cf30403f6018/Screenshot-2.png)

* If we look closely on the first invoice then we can see that the initial payment of ``$ 7118.45`` was calculated twice as the *previous balance & the payment* both, which causes the wrong invoice calculation for future invoices.  
Here, the total of the 1st invoice should be `$ - 2130.95` instead of `$ - 9249.40` & the previous balance of the 2nd invoice also should be `$ - 2130.95` instead of `$ - 9249.40`.


* Here, it seems the same account ledger issue as reported previously. So, linking with #877 , #875 & #881



* Resolution - As we stated earlier also that there may be many scenarios which lead to the account ledger issues. So, as per the best understanding of SAB, there may be 2 methods to resolve the account ledger issues.  

1. We have to manually change all of the invoices for the faulty accounts.  
2. Enhance the account ledger functionality so that we can handle such scenarios.

Any route we take there is a detailed discussion required with Tim & may be Ops.


Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 10, 2018, 01:18

assigned to @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 10, 2018, 11:44

This appears to be the same issue as: #208; which your team stated was solved.

Related to the same issue: #203; #452; #463; #464; #525; #751

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 23, 2018, 15:22

@smasih @pchaudhary @tim.traylor @thomas.gentry 

Do we have any updates?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 24, 2018, 07:11

@kdjstudios 

Account Mindray was resolved using Conversion date on merge however "Microsol Resources" is not although both accounts share similar conversion date issue.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 24, 2018, 13:25

I wrote:

> The support team was able to update the "Mindray" account and believes it is now showing correctly. May you please confirm?
> 
> The support team is still working on the "Microsol" account and we will provide further updates as progress is made.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 25, 2018, 01:40

@kdjstudios 

> The support team was able to update the "Mindray" account and believes it is now showing correctly. May you please confirm?

Sorry my bad, account "Mindray's" ledger issue was resolved and tested on merge, deployment of it's fixed is still pending on production, however it's scheduled for this Saturday.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 5, 2018, 14:17

@smasih @pchaudhary 

Response from Cindy:

> Yes Mindray is showing correct, but as of today and this email Microsol is not.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 6, 2018, 01:55

Hi [Kyle] (@kdjstudios ),  

We investigated the account *Microsol Resources* & found that there are some payments which may be causing the issue.   

![image](/uploads/368122f137dcbd91013d8d42de4f79d5/image.png)


*  The payment of `$ 5600` was created at `08/26/2017 21:12` but the paid date was adjusted to the `08/12/2015`.
*  We also would like to mention that the payment of the same `$ 5600` was reflected to the invoice # `063-3256` having the invoice date of `08/01/2017`.

![image](/uploads/c8d17efe3450463ff8203920a5114967/image.png)

![image](/uploads/a1af1d96e6b3b53605f5bcd939c3e7e9/image.png)

* The same issue with these payments also.

So, in short, there are some payments which may have the wrong paid date. Please confirm with Cindy.

* We are not sure that after fixing these the ledger will be adjusted correct but this is the first thing we got. We'll look further into this after fixing the same.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 6, 2018, 10:17

I wrote:

> Thank you for the update. The support team was able to look into the Microsol account and believes they have found a possible cause. They believe there is possibly payments that may have the wrong paid date. Please see below and the attached screenshot. Please let us know if you can confirm that the paid dates are correct or incorrect?
> 
> - The payment of $ 5600 was created at 08/26/2017 21:12 but the paid date was adjusted to the 08/12/2015.
> - We also would like to mention that the payment of the same $ 5600 was reflected to the invoice # 063-3256 having the invoice date of 08/01/2017.

![image](/uploads/00f781d34b9d00578a42559c7a2d2417/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 6, 2018, 10:50

Cindy wrote:

> These appear to be correct. 

I wrote:

> We do apologize but your response has left us slightly confused.
> 
> Are you confirming that account had a missed payment for over two years? And then the payment was posted two years late and marked it as an 'paid' date of two years prior?
> 
> Or are you confirming that the account does have incorrect payment 'paid' dates? If so, please provide the correct 'paid' dates.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 6, 2018, 11:02

Cindy Wrote:

> Paid date for one is to be 8/12/2015 and one for 8/26/2017.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 6, 2018, 11:08

I wrote:

> We are slightly more confused now. We do not see a date of 8/12/15 anywhere? 
> 
> Let me possibly summarized in detail what we would like to believe you are saying. In the attached screenshot previously we had highlighted a payment that was posted in SAB on 08/26/17; however it was marked to have an effective paid date of roughly two years previous on 12/08/15. Thus, there was a missing payment for two years in SAB, correct?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 6, 2018, 13:52

Cindy Wrote:

> 1) The overall balance of the account of 5425 is accurate. 
>
> 2) The 12/8/2015 payment is correct. There was no payment entered in that month for December 2015 and that is the correct. That payment needed to be posted as it was missing. 

I wrote:

> Thank you for using detail in your description and explaining the account. We will inform the support team.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 25, 2018, 04:27

moving to next

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 27, 2018, 02:25

@kdjstudios 

@pchaudhary will have an update on this by eod.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 27, 2018, 06:57

Hi [Kyle] ( @kdjstudios ),  

We have investigated the account further & found that the invoice number 063-3427 did not carry the correct previous balance due to which this whole thing s occurred. 

![image](/uploads/add111425323583ae19c9c85abf49d52/image.png)

We have to change the previous balance of that invoice which will recalculate all of the further invoices from that invoice & will make the account balance equals to $ -8801.25 as the ledger.

Please confirm with Cindy to move further.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 27, 2018, 11:15

I wrote:

> The support team investigated the account further & found that the invoice number 063-3427 did not carry the correct previous balance over. The support team will have to change the previous balance of that invoice which will recalculate all of the further invoices from that invoice & will make the account balance equals to $ -8801.25. Please provide your approval for the support team to make the correction or if you have any questions or concerns. Thank you and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 27, 2018, 11:42

Cindy wrote:

> approved.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 28, 2018, 24:49

Hi [Kyle] (@kdjstudios ),  

We have successfully changed the invoices & the account balance. Please confirm with Cindy.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 28, 2018, 08:49

I wrote:

> The support team has made adjustments on the account. May you please confirm everything is correct. Thank you and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 9, 2018, 13:03

Closing ticket due to lack of response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 9, 2018, 13:04

closed
